src/backend/common/Logger.cpp

```diff
@@ -0,0 +1,55 @@
+
+#include <common/Logger.hpp>
+#include <common/util.hpp>
+
+#include <array>
+#include <cstdlib>
+#include <string>
+#include <memory>
+
+using std::array;
+using std::make_shared;
+using std::string;
+using std::shared_ptr;
+using std::to_string;
+
+using spdlog::level::trace;
+using spdlog::logger;
+using spdlog::stdout_logger_mt;
+
+namespace common {
+
+#ifdef AF_WITH_LOGGING
+shared_ptr<logger>
+logger_factory(string name) {
+    auto logger = stdout_logger_mt(name);
+    logger->set_pattern("[%n][%t] %v");
+
+    // Log mode
+    string env_var = getEnvVar("AF_TRACE");
+    if(env_var.find_first_of("all") != string::npos ||
+       env_var.find_first_of(name) != string::npos)
+        logger->set_level(trace);
```

Comment:
I couldn't understand what's the effect of trace level, what happens if we don't set trace level ?

What is the default value of `AF_TRACE` environment variable ? If we don't set `AF_MEM`, then trace wouldn't be enabled right ? How will the log look like then ?